src/uct/ib/ud/base/ud_iface.c

```diff
@@ -305,6 +305,27 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, uct_ud_iface_config_t *config)
     return UCS_ERR_INVALID_PARAM;
 }
 
+void uct_ud_iface_complete_init(uct_ud_iface_t *iface, ucs_notifier_chain_func_t progress_cb)
+{
+    ucs_status_t status;
+
+    /* TODO: add progress on first ep creation */
+    uct_ud_enter(iface); 
```

Comment:
@alex-mikheev adding here the what we talked f2f- we could avoid taking the lock here if adding the timer would be the very last operation done in the ud iface initialization